test/e2e/specs/editor/various/embedding.spec.js

```diff
@@ -0,0 +1,310 @@
+
+const EMBED_URLS = [
+	'/oembed/1.0/proxy',
+	`rest_route=${ encodeURIComponent( '/oembed/1.0/proxy' ) }`,
+];
+
+const MOCK_EMBED_WORDPRESS_SUCCESS_RESPONSE = {
+	url: 'https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/',
+	html: '<div class="wp-embedded-content" data-secret="shhhh it is a secret"></div>',
+	type: 'rich',
+	provider_name: 'WordPress',
+	provider_url: 'https://wordpress.org',
+	version: '1.0',
+};
+
+const MOCK_EMBED_RICH_SUCCESS_RESPONSE = {
+	url: 'https://twitter.com/notnownikki',
+	html: '<p>Mock success response.</p>',
+	type: 'rich',
+	provider_name: 'Twitter',
+	provider_url: 'https://twitter.com',
+	version: '1.0',
+};
+
+const MOCK_EMBED_PHOTO_SUCCESS_RESPONSE = {
+	url: 'https://cloudup.com/cQFlxqtY4ob',
+	html: '<p>Mock success response.</p>',
+	type: 'photo',
+	provider_name: 'Cloudup',
+	provider_url: 'https://cloudup.com',
+	version: '1.0',
+};
+
+const MOCK_EMBED_VIDEO_SUCCESS_RESPONSE = {
+	url: 'https://www.youtube.com/watch?v=lXMskKTw3Bc',
+	html: '<iframe width="16" height="9"></iframe>',
+	type: 'video',
+	provider_name: 'YouTube',
+	provider_url: 'https://youtube.com',
+	version: '1.0',
+};
+
+const MOCK_BAD_EMBED_PROVIDER_RESPONSE = {
+	url: 'https://twitter.com/thatbunty',
+	html: false,
+	provider_name: 'Embed Provider',
+	version: '1.0',
+};
+
+const MOCK_CANT_EMBED_RESPONSE = {
+	provider_name: 'Embed Handler',
+	html: '<a href="https://twitter.com/wooyaygutenberg123454312">https://twitter.com/wooyaygutenberg123454312</a>',
+};
+
+const MOCK_BAD_WORDPRESS_RESPONSE = {
+	code: 'oembed_invalid_url',
+	message: 'Not Found',
+	data: {
+		status: 404,
+	},
+	html: false,
+};
+
+test.use( {
+	embedUtils: async ( { page, editor }, use ) => {
+		await use( new EmbedUtils( { page, editor } ) );
+	},
+} );
+
+test.describe( 'Embedding content', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should render embeds in the correct state', async ( {
+		editor,
+		embedUtils,
+	} ) => {
+		await embedUtils.interceptRequests( {
+			'https://twitter.com/notnownikki': MOCK_EMBED_RICH_SUCCESS_RESPONSE,
+			'https://twitter.com/wooyaygutenberg123454312':
+				MOCK_CANT_EMBED_RESPONSE,
+			'https://wordpress.org/gutenberg/handbook/':
+				MOCK_BAD_WORDPRESS_RESPONSE,
+			'https://twitter.com/thatbunty': MOCK_BAD_EMBED_PROVIDER_RESPONSE,
+			'https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/':
+				MOCK_EMBED_WORDPRESS_SUCCESS_RESPONSE,
+			'https://www.youtube.com/watch?v=lXMskKTw3Bc':
+				MOCK_EMBED_VIDEO_SUCCESS_RESPONSE,
+			'https://cloudup.com/cQFlxqtY4ob':
+				MOCK_EMBED_PHOTO_SUCCESS_RESPONSE,
+		} );
+
+		const currenEmbedBlock = editor.canvas
+			.getByRole( 'document', { name: 'Block' } )
+			.last();
+
+		await embedUtils.insertEmbed( 'https://twitter.com/notnownikki' );
+		await expect(
+			currenEmbedBlock.locator( 'iframe' ),
+			'Valid embed. Should render valid element.'
```

Comment:
I didn't know that we could specify an optional message as the second argument 😄